bin/testObservability/helper/helper.js

```diff
@@ -766,13 +767,44 @@ const getLocalSessionReporter = () => {
 }
 
 const cleanupTestObservabilityFlags = (rawArgs) => {
-  const newRawArgs = [];
-  for(let idx=0; idx<rawArgs.length; idx++) {
-    if(!['--disable-browserstack-automation', '--disable-test-observability'].includes(rawArgs[idx])) {
-      newRawArgs.push(rawArgs[idx]);
+  const newArgs = [];
+  const aliasMap = Object.keys(runOptions).reduce( (acc, key) => {
+    const curr = runOptions[key];
+    if (curr.alias)	 {
+      const aliases = Array.isArray(curr.alias) ? curr.alias : [curr.alias] 
+      for (const alias of aliases) {
+        acc[alias] = curr;
+      }
+    }
+    return acc;
+  }, {})
+
+  const cliArgs = {
+    ...runOptions,
+    ...aliasMap
+  }
+
+  // these flags are present in cypress too, but in some the same cli and
+  // cypress flags have different meaning. In that case, we assume user has
+  // given cypress related args
+  const retain = ['c', 'p', 'b', 'o', 's', 'specs', 'spec']
+
+  for (let i = 0;i < rawArgs.length;i++) {
+    const arg = rawArgs[i];
+    const next = i + 1 < rawArgs.length ? rawArgs[i+1] : ''
+    if (arg.startsWith('-')) {
+      const argName = arg.length > 1 && arg[1] == '-' ? arg.slice(2) : arg.slice(1);
+      // If this argument belongs to cli, we omit it and its value
+      if (cliArgs[argName] && !retain.includes(argName)) {
+        if (cliArgs[argName].type && cliArgs[argName].type !== 'boolean' && !next.startsWith('-')) {
```

Comment:
Please explain the logic implemented here

Reply:
Added comment